clang/test/CIR/CodeGen/lambda.cpp

```diff
@@ -285,3 +288,92 @@ int g3() {
+  int bar() { return [this] { return a; }(); }
+};
+// A's default ctor
+// CHECK: cir.func linkonce_odr @_ZN1AC1Ev(%arg0: !cir.ptr<!ty_A>
+
+// lambda operator() in foo()
+// CHECK: cir.func lambda linkonce_odr @_ZZN1A3fooEvENKUlvE_clEv([[ARG:%.*]]: !cir.ptr<!ty_anon2E7_>
+// CHECK: [[ARG_ADDR:%.*]] = cir.alloca !cir.ptr<!ty_anon2E7_>, !cir.ptr<!cir.ptr<!ty_anon2E7_>>, ["this", init] {alignment = 8 : i64}
+// CHECK: cir.store [[ARG]], [[ARG_ADDR]] : !cir.ptr<!ty_anon2E7_>, !cir.ptr<!cir.ptr<!ty_anon2E7_>>
+// CHECK: [[CLS_ANNO7:%.*]] = cir.load [[ARG_ADDR]] : !cir.ptr<!cir.ptr<!ty_anon2E7_>>, !cir.ptr<!ty_anon2E7_>
+// CHECK: [[STRUCT_A:%.*]] = cir.get_member [[CLS_ANNO7]][0] {name = ""} : !cir.ptr<!ty_anon2E7_> -> !cir.ptr<!ty_A>
+// CHECK: [[a:%.*]] = cir.get_member [[STRUCT_A]][0] {name = "a"} : !cir.ptr<!ty_A> -> !cir.ptr<!s32i> loc(#loc70)
+// CHECK: cir.return {{%.*}} : !s32i
+
+// LLVM: define {{.*}}@_ZZN1A3fooEvENKUlvE_clEv(ptr{{.*}}[[ARG:%.*]])
+// LLVM: [[ARG_ADDR:%.*]]  = alloca ptr, i64 1, align 8
+// LLVM: [[RET:%.*]] = alloca i32, i64 1, align 4
+// LLVM: store ptr [[ARG]], ptr [[ARG_ADDR]], align 8
+// LLVM: [[CLS_ANNO7:%.*]] = load ptr, ptr [[ARG_ADDR]], align 8
+// LLVM: [[STRUCT_A:%.*]] = getelementptr %class.anon.7, ptr [[CLS_ANNO7]], i32 0, i32 0
+// LLVM: [[a:%.*]] = getelementptr %struct.A, ptr [[STRUCT_A]], i32 0, i32 0
+// LLVM: [[TMP0:%.*]] = load i32, ptr [[a]], align 4
+// LLVM: store i32 [[TMP0]], ptr [[RET]], align 4
+// LLVM: [[TMP1:%.*]] = load i32, ptr [[RET]], align 4
+// LLVM: ret i32 [[TMP1]]
+
+// A::foo()
+// CHECK: cir.func linkonce_odr @_ZN1A3fooEv(%arg0: !cir.ptr<!ty_A>{{.*}}-> !s32i
+// CHECK: [[THIS_ARG:%.*]] = cir.alloca !ty_anon2E7_, !cir.ptr<!ty_anon2E7_>, ["ref.tmp0"] {alignment = 4 : i64}
+// CHECK: cir.call @_ZZN1A3fooEvENKUlvE_clEv([[THIS_ARG]]) : (!cir.ptr<!ty_anon2E7_>) -> !s32i
+
+// LLVM-LABEL: _ZN1A3fooEv
+// LLVM: [[this_in_foo:%.*]] =  alloca %class.anon.7, i64 1, align 4
+// LLVM: call i32 @_ZZN1A3fooEvENKUlvE_clEv(ptr [[this_in_foo]])
+
+// lambda operator() in bar()
+// CHECK: cir.func lambda linkonce_odr @_ZZN1A3barEvENKUlvE_clEv([[ARG2:%.*]]: !cir.ptr<!ty_anon2E8_>
+// CHECK: [[ARG2_ADDR:%.*]] = cir.alloca !cir.ptr<!ty_anon2E8_>, !cir.ptr<!cir.ptr<!ty_anon2E8_>>, ["this", init] {alignment = 8 : i64}
+// CHECK: cir.store [[ARG2]], [[ARG2_ADDR]] : !cir.ptr<!ty_anon2E8_>, !cir.ptr<!cir.ptr<!ty_anon2E8_>>
+// CHECK: [[CLS_ANNO8:%.*]] = cir.load [[ARG2_ADDR]] : !cir.ptr<!cir.ptr<!ty_anon2E8_>>, !cir.ptr<!ty_anon2E8_>
+// CHECK: [[STRUCT_A_PTR:%.*]] = cir.get_member [[CLS_ANNO8]][0] {name = ""} : !cir.ptr<!ty_anon2E8_> -> !cir.ptr<!cir.ptr<!ty_A>>
+// CHECK: [[STRUCT_A:%.*]] = cir.load [[STRUCT_A_PTR]] : !cir.ptr<!cir.ptr<!ty_A>>, !cir.ptr<!ty_A>
+// CHECK: [[a:%.*]] = cir.get_member [[STRUCT_A]][0] {name = "a"} : !cir.ptr<!ty_A> -> !cir.ptr<!s32i> loc(#loc70)
+// CHECK: cir.return {{%.*}} : !s32i
+
+// LLVM: define {{.*}}@_ZZN1A3barEvENKUlvE_clEv(ptr{{.*}}[[ARG2:%.*]])
```

Comment:
Function labels should all use `CHECK-LABEL` like stuff, just noticed from one of your previous PR you also used `{{.*}}i32 @"_ZZ2g2vENK3$_0clEv"(ptr [[THIS:%.*]])`, please fix those while here, it's very confusing. Please fix that for both LLVM and CIR function label check lines.

Reply:
sure, will change it. I was doing this just so we could test argument of the function is actually used as it should be, but probably not necessary.